ebos/eclproblem.hh

```diff
@@ -3020,115 +3020,114 @@ private:
+                                if (elemIdx >= 0)
                                     (*data)[elemIdx][compIdx] = rate;
-                                }
                             }
                         }
-                    } else if (type == "FREE") {
-                        std::vector<bool>* data = 0;
-                        if (direction == "X-")
-                            data = &freebcXMinus_;
-                        else if (direction == "X")
-                            data = &freebcX_;
-                        else if (direction == "Y-")
-                            data = &freebcYMinus_;
-                        else if (direction == "Y")
-                            data = &freebcY_;
-                        else if (direction == "Z-")
-                            data = &freebcZMinus_;
-                        else if (direction == "Z")
-                            data = &freebcZ_;
-                        else
-                            throw std::logic_error("invalid direction for BC");
-
-                        for (int i = i1; i <= i2; ++i) {
-                            for (int j = j1; j <= j2; ++j) {
-                                for (int k = k1; k <= k2; ++k) {
-                                    std::array<int, 3> tmp = {i,j,k};
-                                    size_t elemIdx = cartesianToCompressedElemIdx[vanguard.cartesianIndex(tmp)];
-                                    (*data)[elemIdx] = true;
-                                }
+                    }
+                } else if (type == BCType::FREE) {
+                    std::vector<bool>* data = nullptr;
+                    switch (bcface.dir) {
+                    case FaceDir::XMinus:
+                        data = &freebcXMinus_;
+                        break;
+                    case FaceDir::XPlus:
+                        data = &freebcX_;
+                        break;
+                    case FaceDir::YMinus:
+                        data = &freebcYMinus_;
+                        break;
+                    case FaceDir::YPlus:
+                        data = &freebcY_;
+                        break;
+                    case FaceDir::ZMinus:
+                        data = &freebcZMinus_;
+                        break;
+                    case FaceDir::ZPlus:
+                        data = &freebcZ_;
+                        break;
+                    }
+
+                    for (int i = bcface.i1; i <= bcface.i2; ++i) {
+                        for (int j = bcface.j1; j <= bcface.j2; ++j) {
+                            for (int k = bcface.k1; k <= bcface.k2; ++k) {
+                                std::array<int, 3> tmp = {i,j,k};
+                                auto elemIdx = cartesianToCompressedElemIdx[vanguard.cartesianIndex(tmp)];
+                                if (elemIdx >= 0)
+                                    (*data)[elemIdx][compIdx] = true;
```

Comment:
I **think** this will be a build problem.  The original code used
```C++
(*data)[elemIdx] = true;
```
but the new code uses
```C++
(*data)[elemIdx][compIdx] = true;
```

Reply:
Fixed